src/CMakeLists.txt

```diff
@@ -43,7 +43,8 @@ set(HIPSYCL_SOURCES
   runtime/hip/hip_queue.cpp
   runtime/hip/hip_allocator.cpp
   runtime/hip/hip_device_manager.cpp
-  runtime/hw_model/memcpy.cpp)
+  runtime/hw_model/memcpy.cpp
+  runtime/common/serialization.cpp)
```

Comment:
Why do we need to put `serialization.cpp` in a new `common` subdirectory? Can't we just have it in `runtime/serialization.cpp`? There's enough generic stuff already in that directory.

Reply:
would you also prefer the headers without their own dierectory?